server/src/main/java/org/opensearch/node/NodeResourceUsageStats.java

```diff
@@ -107,4 +116,29 @@ public void setIoUsageStats(IoUsageStats ioUsageStats) {
     public long getTimestamp() {
         return timestamp;
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(nodeId);
+        builder.field(TIMESTAMP, timestamp);
+        builder.field(CPU_UTILIZATION_PERCENT, String.format(Locale.ROOT, "%.1f", cpuUtilizationPercent));
+        builder.field(MEMORY_UTILIZATION_PERCENT, String.format(Locale.ROOT, "%.1f", memoryUtilizationPercent));
+        if (ioUsageStats != null) {
+            builder.field(IO_USAGE_STATS, ioUsageStats);
+        }
+        builder.endObject();
+        return builder;
+    }
```

Comment:
We likely don't need the XContent changes if this is expected to be consumed internally?